state/modelgeneration_test.go

```diff
@@ -208,9 +210,60 @@ func (s *generationSuite) TestCommitNoChangesEffectivelyAborted(c *gc.C) {
 
 // TODO (manadart 2019-03-21): Tests for abort.
 
+func (s *generationSuite) TestBranchCharmConfigDeltas(c *gc.C) {
+	gen := s.setupAssignAllUnits(c)
+	c.Assert(gen.Config(), gc.HasLen, 0)
+
+	current := state.GetPopulatedSettings(map[string]interface{}{
```

Comment:
Worth adding a test for the escaping/unescaping? Just having a key with a '.' or '$' would be good.